### ISSUE TITLE
Question: Running multiple scripts via cloud-init

### ISSUE THREAD

**Issue by elwyne-mabanglo** [oracle/terraform-provider-oci#563]
### Terraform Version

<details>
# Run this command to get the terraform version:

Terraform v0.11.7
+ provider.oci v2.1.13
+ provider.template v1.0.0
</details>

### OCI Provider Version

<details>

terraform-provider-oci_v2.1.13

</details>

### Description:
Is it possible to run multiple scripts via cloud-init? 

Example below:

### Terraform Plan

`metadata = {`
`       ssh_authorized_keys = "${file(var.ssh_public_key)}"`
`       user_data = "[${base64encode(file("./scripts1.sh"))},"${base64encode(file("./scripts2.sh"))}"]"`
`}`

**Comment by alexng-canuck** [comment]
Hi @elwyne-mabanglo 

Specifying multiple scripts into `user_data` is not supported. The `user_data` value must always be a single base64 encoded string value.

Can you help us understand the use case for this? Could you combine all the scripts into one master script before passing it into `user-data`?

Thanks,
Alex

**Comment by elwyne-mabanglo** [comment]
Hi @alexng-canuck 

We have a number of scripts that we would like to run in isolation from each other, combining the scripts may cause some conflicts.

Hi @hyder 

I'll have a look at this, thanks hopefully this may resolve this issue. 
 
Thanks,
Elwyne

**Comment by elwyne-mabanglo** [comment]
Hi @hyder,

I tried the following code below and haven't had much success getting it to work.

The two scripts just have "echo 'This is a test' > foo.txt", to see if it works. Is the code I'm using correct?

data "template_file" "script1" {
  template = "${file("./script1.sh")}"
}

data "template_file" "script2" {
  template = "${file("./script2.sh")}"
}

data "template_cloudinit_config" "config" {
  gzip          = false
  base64_encode = true

  part {
    filename     = "script1.sh"
    content_type = "text/part-handler"
    content      = "${data.template_file.script1.rendered}"
  }

  part {
    filename     = "script2.sh"
    content_type = "text/part-handler"
    content      = "${data.template_file.script2.rendered}"
  }  
}

**Comment by bjd2385** [comment]
@elwyne-mabanglo did that approach work? I've actually come across this same problem and was thinking of that very approach.

**Comment by hyder** [comment]
A couple of things:

1. You should use the templatefile function as the template provider is deprecated. See here: https://github.com/oracle-terraform-modules/terraform-oci-bastion/blob/main/locals.tf#L16 for an example. See here https://lmukadam.medium.com/a-little-update-and-some-terraform-goodies-f104edc5e4c8 under **Terraform goodies**
```hcl
  # in locals
  # cloud init template file
  mycloudinit_template = "${path.module}/cloudinit/operator.template.yaml"
   
  # script1 - no arguments to the script
   script1_template = base64gzip(
    templatefile("${path.module}/scripts/script1.template.sh", {})
  ) 
 # script2 passes an variable to the sript
  script2_template = base64gzip(
    templatefile("${path.module}/scripts/script2.template.sh",
      {
        ol = var.operating_system_version
      }
    )
  )
```
2. You can then pass them to cloud init as follows

```hcl
data "cloudinit_config" "bastion" {
  gzip          = true
  base64_encode = true

  part {
    filename     = "mycloudinit.yaml"
    content_type = "text/cloud-config"
    content = templatefile(
      local.mycloudinit_template, {
        script1_content = local.script1_template,
        script2_content = local.script2_template
      }
    )
  }
}
```
3. your cloud init must also include a corresponding parameter for the arguments above

```hcl
#cloud-config
package_upgrade: ${upgrade_bastion}
timezone: ${bastion_timezone}

write_files:
# setup script
  - path: "/home/opc/script1.sh"
    permissions: "0700"
    encoding: "gzip+base64"
    content: |
      ${script1_content}
  - path: "/home/opc/script2.sh"
    permissions: "0700"
    encoding: "gzip+base64"
    content: |
      ${script2_content}
```

Hope that helps.

**Comment by ring2003** [comment]
I solve it like below:

TLDR; just trim the $BOUNDARY:
`replace(data.cloudinit_config.basics.rendered, "--MIMEBOUNDARY--","")`

i.e.:

```hcl
data "cloudinit_config" "main" {
    gzip = false
    base64_encode = false

    // 01 - Basics
    part {
        content_type = "text/cloud-config;charset=utf-8"
        content = replace(data.cloudinit_config.basics.rendered, "--MIMEBOUNDARY--","")
        merge_type = "list(append)+dict(recurse_array)+str()"
    }

    // 02 - Extras
    part {
        content_type = "text/cloud-config;charset=utf-8"
        content = join("\n",[for i in data.cloudinit_config.extras: replace(i.rendered, "--MIMEBOUNDARY--","")])
        merge_type = "list(append)+dict(recurse_array)+str()"
    }
}
data "cloudinit_config" "basics" {
    gzip = false
    base64_encode = false 
    // ...
}
data "cloudinit_config" "extras" {
    count = var.cloudinit.extras.enabled ? length(var.cloudinit.extras.scripts) : 0
    gzip = false
    base64_encode = false
    part {
        content_type = "text/x-shellscript;charset=utf-8"
        content = file(var.cloudinit.extras.scripts[count.index])
    }
}
```
